### DISCUSSION TITLE
Why do we need to give a provider no matter what?

### DISCUSSION THREAD

**Discussion by johnnyshankman** [wevm/wagmi#1616] [Q&A]
If i want the user to bring their own provider via Metamask, why am I required to still pass in `publicProvider()` into my `configureChains()` call? I will never use this provider at all.

**Comment by jxom** [comment]
Because using the wallet as an RPC can be dangerous.

Main reason is because some wallets restrict what RPC methods they provide access to.

See here: https://github.com/wagmi-dev/wagmi/discussions/950#discussioncomment-4200440
